### PR TITLE
many: merge 2.45.2 fixes back into master

### PR DESCRIPTION
**Pull request by mvo5**
This PR merges the 2.45.2 fixes back into master.

### CODE REVIEW

File: tests/main/xdg-open/task.yaml [hunk 1]
```diff
@@ -85,11 +84,6 @@ execute: |
     ensure_xdg_open_output "help:snapcraft"
     ensure_xdg_open_output "apt:snapcraft"
     ensure_xdg_open_output "zoommtg://snapcraft.io"
-    ensure_xdg_open_output "slack://5NAPPY111/magic-login/bcaf81ee-07b1-4362-9c09-ff46bd6e1bb9"
-    ensure_xdg_open_output "msteams://snapcraft.io"
-
-    # Ensure XDG_DATA_DIRS was prefixed with snap specific location
-    test "$(tail -1 /tmp/xdg-open-output)" = "XDG_DATA_DIRS=$SNAP_MOUNT_DIR/test-snapd-desktop/current/usr/share:/usr/share"
```
Comment (anonymouse64) [inline tests/main/xdg-open/task.yaml:92]: is this diff right?

Comment (mvo5) [reply tests/main/xdg-open/task.yaml:92]: it is not :( thank you

Comment (mvo5) [reply tests/main/xdg-open/task.yaml:92]: I pushed an update but please double check, this test has changed quite a bit

**Comment by mvo5** [comment]
I submited individual PRs now to make this less unwieldy in 9022-9026

**Comment by pedronis** [comment]
marking blocked until the individual PRs are merged

**Comment by pedronis** [comment]
I merged master that had most of the commits merged piece-wise, and this is now unblocked

**Comment by anonymouse64** [comment]
The test failure on UC16 here is from https://github.com/snapcore/snapd/pull/8861:

```
##[error]2020-07-23 10:25:34 Error executing google:ubuntu-core-16-64:tests/core/snapd16 (jul231004-223201) : 
-----
+ '[' 0 = 0 ']'
+ echo 'No snapd snap is installed'
No snapd snap is installed
+ not snap list snapd
error: no matching snaps installed
+ echo 'Ensure the snapd snap can be installed'
Ensure the snapd snap can be installed
+ snap install snapd-from-core.snap --dangerous
error: cannot perform the following tasks:
- Make snap "snapd" (unset) available to the system (open /etc/dbus-1/session.d/snapd.session-services.conf.TKS7tBf1GgpQ~: read-only file system)
-----
```

so I think this is ready to merge

**Pull request merged**